src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/RazorDocumentServiceProvider.cs

```diff
@@ -68,6 +69,22 @@ public RazorDocumentServiceProvider(DynamicDocumentContainer documentContainer)
                 return (TService)_excerptService;
             }
 
+            if (typeof(TService) == typeof(IRazorDocumentPropertiesService))
```

Comment:
What's the context behind this manual handling of each service? This seems like a time for some form of Dependency Injection but I assume there's something working against that?

Reply:
I wasn't involved in the document services design so I can't give you a 100% accurate answer but my guess would be that not having a service provider enables Roslyn to query document services without requiring a backing service provider for each document (may be expensive).